drivers/spi/spi_sam0.c

```diff
@@ -657,6 +671,9 @@ static int spi_sam0_init(const struct device *dev)
 			  | GCLK_CLKCTRL_ID(cfg->gclk_id);
 #endif
 
+	regs->CTRLA.bit.SWRST = 1;
+	wait_synchronization(regs);
+
```

Comment:
I see two things in your PR:
1- Add the software reset at init, which makes sense to me.
2- Handle the 32-bit len extension.

In this case, I think this part should be in their own commit.

Reply:
I will split these changes!